src/platform/K32W/Logging.cpp

```diff
@@ -21,60 +27,59 @@
 static bool isLogInitialized;
 extern uint8_t gOtLogUartInstance;
 extern "C" void K32WWriteBlocking(const uint8_t * aBuf, uint32_t len);
+extern "C" uint32_t otPlatAlarmMilliGetNow(void);
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
-void GetMessageString(char * buf, uint8_t chipCategory, uint8_t otLevelLog)
+void GetMessageString(char * buf, uint8_t bufLen, const char * module, uint8_t category)
 {
-    if (chipCategory != kLogCategory_None)
+    int writtenLen = 0;
+    const char * categoryString;
+
+    /* bufLen must accommodate the length of the timestamp + length of the category +
+     * length of the module + 2 bytes for the module's brackets +
+     * 1 byte for the terminating character.
+     */
+    assert(bufLen >= (timestamp_max_len_bytes + category_max_len_bytes + (strlen(module) + 2) + 1));
+
+    writtenLen = snprintf(buf, bufLen, "[%lu]", otPlatAlarmMilliGetNow());
+    bufLen -= writtenLen;
+    buf += writtenLen;
+
+    if (category != kLogCategory_None)
     {
-        switch (chipCategory)
+        switch (category)
         {
         case kLogCategory_Error:
-            memcpy(buf, "[Error]", 7);
+            categoryString = "E";
             break;
         case kLogCategory_Progress:
-        default:
-            memcpy(buf, "[Progress]", 10);
+            categoryString = "P";
             break;
         case kLogCategory_Detail:
-            memcpy(buf, "[Debug]", 7);
+            categoryString = "D";
             break;
-        }
-    }
-
-    if (otLevelLog != OT_LOG_LEVEL_NONE)
-    {
-        switch (otLevelLog)
-        {
-        case OT_LOG_LEVEL_CRIT:
-            memcpy(buf, "[Error]", 7);
-            break;
-        case OT_LOG_LEVEL_WARN:
-            memcpy(buf, "[Warn]", 6);
-            break;
-        case OT_LOG_LEVEL_NOTE:
-        case OT_LOG_LEVEL_INFO:
         default:
-            memcpy(buf, "[Info]", 6);
-            break;
-        case OT_LOG_LEVEL_DEBG:
-            memcpy(buf, "[Debug]", 7);
-            break;
+            categoryString = "U";
         }
+
+        writtenLen = snprintf(buf, bufLen, "[%s]", categoryString);
+        bufLen -= writtenLen;
+        buf += writtenLen;
     }
+
+    writtenLen = snprintf(buf + writtenLen, bufLen, "[%s]", module);
```

Comment:
Didn't we already add writtenLen ?

Reply:
@bzbarsky-apple @doru91

Reply:
Er, yes, so we did.  Good catch!  @doru91 please fix?

Reply:
Thanks @mspang @bzbarsky-apple . I opened a new PR: https://github.com/project-chip/connectedhomeip/pull/8471 for fixing this issue as this PR have been merged.

Reply:
Fixed in https://github.com/project-chip/connectedhomeip/pull/8471